src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +154,555 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
   }
 }
 
+TEST_F(SetAppIconRequestTest, OnEvent_InvalidEnum_ReceivedUnknownEvent) {
+  MessageSharedPtr msg_vr = CreateFullParamsUISO();
+
+  utils::SharedPtr<SetAppIconRequest> req_vr =
+      CreateCommand<SetAppIconRequest>(msg_vr);
+
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+
+  Event event(hmi_apis::FunctionID::INVALID_ENUM);
+  event.set_smart_object(*msg);
+
+  req_vr->on_event(event);
+}
+
+TEST_F(SetAppIconRequestTest, OnEvent_NonExistentApp_NullPointer) {
```

Comment:
@BSolonenko The same here . Can not see any checks

Reply:
@AByzhynar Expectation added in https://github.com/smartdevicelink/sdl_core/pull/1974/commits/d3e063fe7994d5ed83665c9492a000bb0dc47552